codalab/worker/worker.py

```diff
@@ -259,13 +265,18 @@ def start(self):
                     self.terminate = True
             except Exception:
                 self.last_checkin_successful = False
+                capture_exception()
                 traceback.print_exc()
                 if self.exit_on_exception:
-                    logger.error('Encountered exception, terminating the worker...')
+                    logger.error(
+                        'Encountered exception, terminating the worker after sleeping for 5 minutes...'
+                    )
                     self.terminate = True
+                    # Sleep for 5 minutes
+                    time.sleep(5 * 60)
```

Comment:
Why sleep 5 min?

Reply:
https://github.com/codalab/codalab-worksheets/issues/2720#issue-674801797 , i thought your rationale was to mitigate infinite loops

Reply:
Hm, I was thinking the 5 minutes would replace the sleeping for 1 hour logic.

Reply:
even in the case where workers don't exit on exception (this isn't the default)?